pyproject.toml

```diff
@@ -79,7 +79,10 @@ gcsfs = { version = ">=2023.1.0,<2024.1.0", optional = true }
 psycopg2-binary = { version = ">=2.9.6", optional = true }
 sqlalchemy = { version = "^2.0.18", optional = true }
 getdaft = { version = ">=0.2.12", optional = true }
-numpy = { version = "^1.22.4", optional = true }
+numpy = [
+    { version = "1.26.0", python = ">=3.9,<3.13", optional = true },
+    { version = "1.24.4", python = ">=3.8,<3.9", optional = true }
```

Comment:
3.8 is also almost EOL: https://devguide.python.org/versions/

Reply:
https://github.com/apache/iceberg-python/issues/1121

Created issue to track this